### PR TITLE
Try to fix smc resampling

### PR DESCRIPTION
**Pull request by b-studios**
Hey @mm0821 could you look into this again? This is a draft implementing what you suggested. However, since we now keep live particles around for resampling the probability of 1 in the geometric distribution is VERY high. I am not sure this is correct...

### CODE REVIEW

**Comment by b-studios** [comment]
@phischu you also might be interested in this.

File: docs/casestudies/smc.md [hunk 1]
```diff
@@ -56,16 +52,20 @@ A particle consists of its current _weight_ (or "score"), the _age_ (number of r
 and the continuation that corresponds to the remainder of its computation.
 ```
 record Particle(weight: Double, age: Int, cont: Cont[Unit, Unit])
+record Measurement[R](weight: Double, data: R)
+
+record Particles[R](moving: List[Particle], done: List[Measurement[R]])
 ```
 
 Using the `Particle` data type, we can define our SMC handler as follows:
```
Comment (mm0821) [inline docs/casestudies/smc.md:60]: Maybe it is better to write "Using the above data types ...".

**Review by mm0821 — CHANGES_REQUESTED** [review]
With resetting the weights after resampling, this should be correct. The distribution for `biasedGeometric` should be the same as with importance sampling then.

**Comment by b-studios** [comment]
This is how the distribution looks like with 1000 Particles: 
image

However, the new resampling is embarrasingly slow.... :D

**Review by mm0821 — APPROVED** [review]

**Comment by b-studios** [comment]
I'd be very happy to receive any more efficient implementations of resampling ;)

**Pull request merged**